### PR TITLE
chore(pipeline): support SASS alongside LESS

### PR DESCRIPTION
**Pull request by CITguy**
* add `node-sass` dependency
* initialize `src/scss`
  * copy and convert `src/less/vars.less` to `src/scss/_vars.scss`
  * migrate as many `src/less/*` files as possible w/o breaking existing LESS styles
    * `src/less/config.less` -> `src/scss/_config.scss`
    * `src/less/overrides.less` -> `src/scss/_overrides.scss`
    * `src/less/base.less` -> `src/scss/base/_index.scss`
    * `src/less/base/typography.less` -> `src/scss/base/_typography.scss`
    * `src/less/helpers.less` -> `src/scss/helpers/_index.scss`
    * `src/less/helpers/alignment.less` -> `src/scss/helpers/_alignment.scss`
    * `src/less/helpers/wrapping.less` -> `src/scss/helpers/_wrapping.scss`
    * `src/less/reset.less` -> `src/scss/reset/_index.scss`
    * `src/less/reset/html5.less` -> `src/scss/reset/_html5.scss`
* update `yarn compile` to support incrmental migration from LESS to SASS
  * concatenate LESS-compiled styles to the _end_ of SASS-compiled styles
* refactor/reorganize remaining LESS source code
  * move `src/_bundle.less` to `src/less/index.less`
    * organizes all Light DOM LESS styles in one directory
  * remove useless styles from `src/less/base/links.less`
* partially convert some custom element shadow styles to SCSS
  * `src/elements/hx-*/_shadow.less` -> `src/elements/hx-*/_shadow.scss`
  * `<hx-accordion-panel>`
  * `<hx-error>`
* update rollup configuration to support SCSS imports
  * rename `plugins/` to `rollup-utils/`
  * add `rollup-utils/rollup-plugin-sass.js`
  * refactor `rollup-utils/rollup-plugin-less.js`

JIRA: TBD

### LGTM's
- [x] Dev LGTM

### CODE REVIEW

File: rollup-utils/rollup-plugin-sass.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+export default function plugin (options={}) {
+    let cfg = {
+        // defaults
+        include: [ '**/*.scss' ],
+        exclude: [ 'node_modules/**' ],
+        options: {},
+
+        // overrides
+        ...options
+    };
```
Comment (CITguy) [inline rollup-utils/rollup-plugin-sass.js]: break up `cfg` ... unnecessary, given that `options` is being applied with no overrides from function argument

File: scripts/_compile/index.js [hunk 2]
```diff
@@ -42,12 +58,41 @@ async function compileStyles () {
...
+/**
+ * @see https://sass-lang.com/documentation/js-api
+ * @returns {object} - metadata object with 'css' and 'minified' props
+ */
+async function compileSass () {
+    let sassDir = `${CONFIG.sourceDir}/scss`;
+    let cfg = {
+        file: `${sassDir}/index.scss`, // input/entry
+        includePaths: [ sassDir ]
+    }
```
Comment (CITguy) [inline scripts/_compile/index.js]: semicolon

File: scripts/_compile/index.js [hunk 3]
```diff
@@ -42,12 +58,41 @@ async function compileStyles () {
...
+
+    // SASS -> CSS (full)
+    let renderedCSS = await SASS.renderSync({
```
Comment (CITguy) [inline scripts/_compile/index.js]: save to object, _then_ create new var for the stringified css buffer

File: src/scss/_vars.scss [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+// ==== Z-INDEX =====
+$drawer-z-index: 50;
+$menu-z-index: 100;
+$search-assistance-z-index: 100;
+$popover-z-index: 1000;
+$tooltip-z-index: 1100;
+$modal-z-index: 1200;
+
+$space: (
+  "xs": 0.5rem,
```
Comment (CITguy) [inline src/scss/_vars.scss]: can we go unquoted?

File: src/scss/_vars.scss [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+$screen-xl: 90em; // ~1440px
+// Breakpoint Aliases
+$grid-break-wrist: $screen-xs;
+$grid-break-palm: $screen-sm;
+$grid-break-lap: $screen-md;
+$grid-break-desk: $screen-lg;
+$grid-break-wall: $screen-xl;
+
+// ==== FOCUS STATE =====
+$focus-glow: 0 0 4px fade($cyan-700, 50%);
```
Comment (CITguy) [inline src/scss/_vars.scss]: update `fade()` to SASS equivalent

File: src/scss/_vars.scss [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+$focus-glow-disabled: $focus-glow;
+$focus-glow-invalid: 0 0 4px fade($red-900, 50%);
```
Comment (CITguy) [inline src/scss/_vars.scss]: update `fade()` to SASS equivalent

File: src/scss/_vars.scss [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+$yellow-100: #fff9c4;
+$yellow-500: #ffeb3b;
+$yellow-900: #fbc02d;
+// Reds
+$red-100: #ffcdd2;
+$red-500: #f44336;
+$red-900: #d32f2f;
+// Shadow
+$shadow-color: fade($gray-1000, 16%);
+$backdrop-color: fade($gray-1000, 60%);
```
Comment (CITguy) [inline src/scss/_vars.scss]: update to SASS equivalent

File: src/scss/_vars.scss [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+// Yellows
+$yellow-100: #fff9c4;
+$yellow-500: #ffeb3b;
+$yellow-900: #fbc02d;
+// Reds
+$red-100: #ffcdd2;
+$red-500: #f44336;
+$red-900: #d32f2f;
+// Shadow
+$shadow-color: fade($gray-1000, 16%);
```
Comment (CITguy) [inline src/scss/_vars.scss]: update to SASS equivalent

File: src/scss/_vars.scss [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+// ===== FONTS =====
+$font-sans-serif: "Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif;
+$font-monospace: "Roboto Mono", monospace;
+
+// ===== GRID =====
+// Number of columns in the grid.
+$grid-column-count: 12;
+// Padding between columns.
+$grid-gutter-size: 20px;
+$grid-column-unit: (100% / $grid-column-count);
```
Comment (CITguy) [inline src/scss/_vars.scss]: double check syntax with SASS

File: src/scss/_vars.scss (hunk 4 above)
Comment (CITguy) [reply src/scss/_vars.scss]: yes, unquoting keys

File: src/scss/_vars.scss (hunk 9 above)
Comment (CITguy) [reply src/scss/_vars.scss]: we're good! 

evaluates to `8.3333333333%`

File: rollup.config.js [hunk 10]
```diff
@@ -21,60 +39,80 @@ let htmlPlugin = html ({
...
+// import SASS/SCSS files as JS strings
+// see: https://sass-lang.com/documentation/js-api
+let sassPlugin = sass({
+    options: {
+        // TODO: centralize common config settings
+        includePaths: [
+            'src/scss',
+        ],
+        outputStyle: 'compact', // each style block on its own line (reduce generated white space)
+        precision: 4,
```
Comment (CITguy) [inline rollup.config.js]: This is NOT a magic number. This was determined through research to determine the accuracy of a rounded pixel calculation in extreme cases (e.g., 1 column of a 12-column grid at 360px vs 8K resolution).  A precision of 4 was shown to be accurate all the way up to 8K resolutions, whereas a precision of 3 would be accurate up to 4K.  A precision of 3 would be "good enough" but 4 is "perfect" and it _still_reduces the number of decimal places from 8 (thank you LESS) to 4, saving file size.

File: scripts/_compile/index.js [hunk 11]
```diff
@@ -42,12 +60,41 @@ async function compileStyles () {
...
+}
+
+/**
+ * @see https://sass-lang.com/documentation/js-api
+ * @returns {object}
+ * metadata object with 'css' and 'minified' props whose
+ * values are CSS stylesheet strings
+ */
+async function compileScss () {
+    let scssDir = `${CONFIG.sourceDir}/scss`;
```
Comment (100stacks) [inline scripts/_compile/index.js]: may not need this variable.

File: scripts/_generate/index.js [hunk 12]
```diff
@@ -20,26 +21,57 @@ const Markup = require('./markup');
...
-    await writeFile(`${CONFIG.publicDir}/docs.css`, output.css);
-}//generateStyles()
+    return rendered.css;
+}
+
+/**
+ * @returns {string} styles rendered from SASS
+ */
+async function generateScss () {
+    let inFile = path.join(CONFIG.docsDir, 'index.scss');
```
Comment (100stacks) [inline scripts/_generate/index.js]: do we need this variable?

**Pull request merged**